manim/mobject/svg/svg_mobject.py

```diff
@@ -124,36 +125,36 @@ def _ensure_valid_file(self):
         if self.file_name is None:
             raise Exception("Must specify file for SVGMobject")
 
-        if os.path.exists(self.file_name):
+        if self.file_name.exists():
             self.file_path = self.file_name
             return
 
-        relative = os.path.join(os.getcwd(), self.file_name)
-        if os.path.exists(relative):
+        relative = Path.cwd() / self.file_name
+        if relative.exists():
             self.file_path = relative
             return
 
         possible_paths = [
-            os.path.join(config.get_dir("assets_dir"), self.file_name),
-            os.path.join(config.get_dir("assets_dir"), self.file_name + ".svg"),
-            os.path.join(config.get_dir("assets_dir"), self.file_name + ".xdv"),
-            self.file_name,
-            self.file_name + ".svg",
-            self.file_name + ".xdv",
+            config.get_dir("assets_dir") / self.file_name,
+            config.get_dir("assets_dir") / self.file_name + ".svg",
+            config.get_dir("assets_dir") / self.file_name + ".xdv",
+            self.file_path,
+            self.file_path / ".svg",
+            self.file_path / ".xdv",
```

Comment:
```suggestion
            config.get_dir("assets_dir") / self.file_name,
            config.get_dir("assets_dir") / (str(self.file_name) + ".svg"),
            config.get_dir("assets_dir") / (str(self.file_name) + ".xdv"),
            self.file_path,
            Path(str(self.file_path) + ".svg"),
            Path(str(self.file_path) + ".xdv"),
```

@Kiran-Raj-Dev does this seem better?

Reply:
No, this is not better.

I would suggest you use the [`with_suffix`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix) method of pathlib.Path

Reply:
Ah my bad, saw that method in `help(Path)` but thought it only changed the existing suffix.

New suggestion then:

```suggestion
            config.get_dir("assets_dir") / self.file_name,
            config.get_dir("assets_dir") / self.file_name.with_suffix(".svg"),
            config.get_dir("assets_dir") / self.file_name.with_suffix(".xdv"),
            self.file_path,
            self.file_path.with_suffix(".svg"),
            self.file_path.with_suffix(".xdv"),
```

Reply:
Yup looks much nicer!